dags/analyzer_helper/tests/integration/test_discord_extract_raw_members.py

```diff
@@ -0,0 +1,122 @@
+from hivemind_etl_helpers.src.utils.mongo import MongoSingleton
+
+
+class TestDiscordExtractRawMembers(unittest.TestCase):
+    def setUp(self):
+        self.client = MongoSingleton.get_instance().client
+        self.guild_id = "discord_guild"
+        self.platform_id = "platform_db"
+        self.guild_db = self.client[self.guild_id]
+        self.platform_db = self.client[self.platform_id]
+        self.guild_collection = self.guild_db["guildmemberstest"]
+        self.rawmembers_collection = self.platform_db["rawmemberstest"]
+        self.guild_collection.delete_many({})
+        self.rawmembers_collection.delete_many({})
+
+    def tearDown(self):
+        self.guild_collection.delete_many({})
+        self.rawmembers_collection.delete_many({})
+
+    def test_extract_recompute_true(self):
+        sample_data = [
+            {
+                "discordId": "159985870458322944",
+                "username": "MEE6",
+                "roles": ["1088165451651092635", "1046009276432400435"],
+                "joinedAt": datetime(2023, 3, 22, 18, 21, 0, 870000),
+                "avatar": "b50adff099924dd5e6b72d13f77eb9d7",
+                "isBot": True,
+                "discriminator": "4876",
+                "permissions": "559642693856991",
+                "deletedAt": None,
+                "globalName": None,
+                "nickname": None,
+            },
+            {
+                "discordId": "159985870458322945",
+                "username": "TestUser",
+                "roles": ["1088165451651092636", "1046009276432400436"],
+                "joinedAt": datetime(2023, 3, 23, 18, 21, 0, 870000),
+                "avatar": "a50adff099924dd5e6b72d13f77eb9d8",
+                "isBot": False,
+                "discriminator": "4877",
+                "permissions": "559642693856992",
+                "deletedAt": None,
+                "globalName": "GlobalTestUser",
+                "nickname": "TestNick",
+            },
+        ]
+        self.guild_collection.insert_many(sample_data)
+
+        extractor = DiscordExtractRawMembers(self.guild_id, self.platform_id)
+        result = extractor.extract(recompute=True)
+
+        expected_result = sample_data
+
+        self.assertEqual(result, expected_result)
+
+    def test_extract_recompute_false(self):
+        rawmember_data = [
+            {
+                "discordId": "159985870458322944",
+                "username": "MEE6",
+                "roles": ["1088165451651092635", "1046009276432400435"],
+                "joinedAt": datetime(2023, 3, 22, 18, 21, 0, 870000),
+                "avatar": "b50adff099924dd5e6b72d13f77eb9d7",
+                "isBot": True,
+                "discriminator": "4876",
+                "permissions": "559642693856991",
+                "deletedAt": None,
+                "globalName": None,
+                "nickname": None,
+            }
+        ]
```

Comment:
The provided data format is not the rawmembers collection data format. Please consider updating it to the right data format.

ref link: https://rndadocs.notion.site/Analyzer-Discord-Pipeline-5009512d82a243deb35b3388e17b29c5#:~:text=rawmembers%20collection%20(transformed%20guildmembers)